quinn-proto/src/connection/mod.rs

```diff
@@ -2962,9 +3026,6 @@ impl Connection {
             }).expect("preferred address CID is the first received, and hence is guaranteed to be legal");
         }
         self.peer_params = params;
-        self.path.max_udp_payload_size = self.path.max_udp_payload_size.min(
-            u16::try_from(self.peer_params.max_udp_payload_size.into_inner()).unwrap_or(u16::MAX),
-        );
```

Comment:
Shouldn't MTU detection be aware of the hard limit set by the peer? This might be lower than our configured upper bound.

Reply:
Yes, I got confused because I didn't know whether this was a hard limit... Just pushed the necessary changes to make MTUD aware of it